zerver/models.py

```diff
@@ -1407,27 +1411,31 @@ def get_status_dicts_for_query(query, mobile_user_ids):
         user_statuses = defaultdict(dict) # type: defaultdict[Any, Dict[Any, Any]]
         # Order of query is important to get a latest status as aggregated status.
         for row in query.order_by("user_profile__id", "-timestamp"):
+            timezone = row['user_profile__timezone']
             info = UserPresence.to_presence_dict(
                 row['client__name'],
                 row['status'],
                 row['timestamp'],
                 push_enabled=row['user_profile__enable_offline_push_notifications'],
                 has_push_devices=row['user_profile__id'] in mobile_user_ids,
                 is_mirror_dummy=row['user_profile__is_mirror_dummy'],
+                local_time=get_local_time_as(timezone),
             )
             if not user_statuses.get(row['user_profile__email']):
                 # Applying the latest status as aggregated status for user.
                 user_statuses[row['user_profile__email']]['aggregated'] = {
                     'status': info['status'],
-                    'timestamp': info['timestamp']
+                    'timestamp': info['timestamp'],
+                    'local_time': get_local_time_as(timezone),
```

Comment:
I'm a little hesitant to do the timezone conversion on the backend since presence data can be a minute or two stale; is there a good way to do this on the frontend instead?  Would also consider putting these data in the `realm_user` / `people.js` data set rather than the presence data, since we only need to send that once vs. every minute.